lib/base58.ex

```diff
@@ -36,11 +36,23 @@ defmodule Base58 do
 
   def decode(""), do: "" # return empty string unmodified
   def decode("\0"), do: "" # treat null values as empty
-  def decode(binary), do: decode(binary, 0)
+  def decode(binary) do
+    {zeroes, binary} = handle_leading_zeroes(binary)
+    zeroes <> decode(binary, 0)
+  end
+  def decode("", 0), do: ""
   def decode("", acc), do: :binary.encode_unsigned(acc)
   def decode(<<head, tail::binary>>, acc),
     do: decode(tail, acc * 58 + Enum.find_index(@alnum, &(&1 == head)))
 
+  def handle_leading_zeroes(binary) do
```

Comment:
This function probably doesn't need to be exposed externally. Should we make it a `defp`? 💭 
```suggestion
  defp handle_leading_zeroes(binary) do
```

Reply:
Good catch, done.